### ISSUE TITLE
Rule blocked: CORERULES-9628 - new operator to check that values of specified variable is valid XHTML

### ISSUE THREAD

**Issue by ASL-rmarshall** [cdisc-org/cdisc-rules-engine#897]
**Links to related JIRA Tickets**
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-9628
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-9629

**Rule Information**
- **Standard**: USDM
- **Rule ID**: DDF00085
- **Rule Description**: Narrative content text is expected to be HTML formatted.

**Describe the problem**
The values of some USDM attributes are expected to be represented in XHTML format (see [XHTML Attributes](https://wiki.cdisc.org/display/USDMIGv4/XHTML+Attributes)). We have defined rules to check that the values in these attributes are valid XHTML, but there isn't yet a CORE operator to check this.

The XHTML reference in the IG indicates that custom USDM tags may be used, which means that strict XHTML validation cannot be used (because custom tags are not allowed in XHTML).  It may be that a custom DTD that includes the additional USDM elements/attributes will need to be defined to be used as the basis for the validation.

It may be that this could be implemented as an operation that performs validation and returns a list of error messages, with the `not_empty` operator then used to report when the result of the operation contains at least one message.

**Proposed rule logic**
```yaml
Check:
  - any:
    - name: $xhtml_errors
      operator: not_empty
Operation:
  - name: text
    operator: get_xhtml_errors
    id: $xhtml_errors 
```
- Excel test dataset contain `text` attribute intended to be in XHTML format: [NarrativeContent.xlsx](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM3l1Qnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--800703a9397682a49018474188792b239daaaeeb/NarrativeContent.xlsx)

**Comment by ASL-rmarshall** [comment]
Still required as there is nothing in JSONata that can do this directly.

**Comment by ASL-rmarshall** [comment]
See [DDF-RA #687](https://github.com/cdisc-org/DDF-RA/issues/687)

**Comment by ASL-rmarshall** [comment]
NarrativeContent dataset in JSON format: [NarrativeContent.json](https://github.com/user-attachments/files/22491625/NarrativeContent.txt)

**Comment by ASL-rmarshall** [comment]
@alexfurmenkov Here's a zip file with the debugging artefacts from a new rule definition and a positive dataset (I'm not expecting it to generate errors): [DDF00187.zip](https://github.com/user-attachments/files/22546032/DDF00187.zip)

I'm a bit unsure of the full range of errors that might be generated, so I haven't created a negative test file yet.  However, I am expecting it to be possible for one data value to generate more than one error message.
